lib/lrama/reporter/states.rb

```diff
@@ -15,273 +15,337 @@ def initialize(itemsets: false, lookaheads: false, solved: false, counterexample
 
       # @rbs (IO io, Lrama::States states) -> void
       def report(io, states)
-        if @counterexamples
-          cex = Counterexamples.new(states)
-        end
+        cex = Counterexamples.new(states) if @counterexamples
```

Comment:
We can move this instance creation into `#report_counterexamples`.

Reply:
Oh, it's wrong. `#report_counterexamples` is called for each `state` then `cex` is needed to allocate outside of the loop.